discovery/discovery.go

```diff
@@ -55,12 +59,65 @@ func (o *orchestratorPool) GetInfos() []common.OrchestratorLocalInfo {
 	return o.infos
 }
 
+func ResolveURL(urlToResolve *url.URL) []*url.URL {
```

Comment:
Should we always add the input `urlToResolve` to `resolvedUrls`, just in case we missed something in the logic below, but the addresses are already equal? That would be safer in terms of backward-compatibility.